x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/queries/VectorDimensionsQueryBuilder.java

```diff
@@ -37,20 +37,21 @@
 import static org.elasticsearch.xpack.ml.queries.TextExpansionQueryBuilder.AllowedFieldType;
 import static org.elasticsearch.xpack.ml.queries.TextExpansionQueryBuilder.PRUNING_CONFIG;
 
-public class WeightedTokensQueryBuilder extends AbstractQueryBuilder<WeightedTokensQueryBuilder> {
+public class VectorDimensionsQueryBuilder extends AbstractQueryBuilder<VectorDimensionsQueryBuilder> {
```

Comment:
The only purpose of this query is to allow `TextExpansionQueryBuilder.java` to continue to exist. I am not sure we should even rename this.

Reply:
We're actively using this in Rally so we can't just get rid of it until there's a viable alternative that's live. I intend to keep it for now and remove it at a later date after we can start replacing those use cases with this API call.

Reply:
I'm OK with keeping it as `WeightedTokensQueryBuilder` until it no longer needs to exist. The functionality though will continue to exist, per discussions we're having with @Mikep86